### PR TITLE
Add fallback command for micromamba

### LINKED ISSUE #4161: `useMicromamba` option doesn't work - generates Conda instructions instead.
### Expected behavior and actual behavior

According to the docs, the [useMicromamba](https://www.nextflow.io/docs/latest/config.html?highlight=usemicromamba#scope-conda) directive should cause Nextflow to use Micromamba instead of Conda.

When using this option, the nextflow process fails to run, because the generated `.command.run` file still uses `conda`.

### Steps to reproduce the problem

On a machine with micromamba installed, but not conda:

```
git clone https://gist.github.com/jemma-nelson/921e3a8aa32efe22bd73fbe1cbdfd09b
cd 921e3a8aa32efe22bd73fbe1cbdfd09b/
bash run.sh
```

### Program output 

The .nextflow.log file, as well as the generated .command.* files are in the gist referenced above. (e.g: https://gist.github.com/jemma-nelson/921e3a8aa32efe22bd73fbe1cbdfd09b#file-nextflow-log)

Program output:

```
N E X T F L O W  ~  version 23.04.2
Launching `main.nf` [nostalgic_ardinghelli] DSL2 - revision: e1ee91cb36
[-        ] process > greet -
Creating env using micromamba: /home/nelsonjs/tmp/repro-micromamba/test-repro/921e3a8aa32efe22bd73fbe1cbdfd09b/env.yaml [cache executor >  local (1)                                                                                                          [e2/667fd3] process > greet (1) [  0%] 0 of 1
executor >  local (1)                                                                                                          [e2/667fd3] process > greet (1) [  0%] 0 of 1                                                                                  executor >  local (1)                                                                                                          [e2/667fd3] process > greet (1) [100%] 1 of 1, failed: 1 ✘                                                                     Creating env using micromamba: /home/nelsonjs/tmp/repro-micromamba/test-repro/921e3a8aa32efe22bd73fbe1cbdfd09b/env.yaml [cache /home/nelsonjs/tmp/repro-micromamba/test-repro/921e3a8aa32efe22bd73fbe1cbdfd09b/work/conda/base-bda312ce1312db01eb3e5e99160dbdb1]
ERROR ~ Error executing process > 'greet (1)'

Caused by:
  Process `greet (1)` terminated with an error exit status (1)

Command executed:

  echo hello world

Command exit status:
  1

Command output:
  (empty)

Command error:
  .command.run: line 119: conda: command not found
  .command.run: line 119: /bin/activate: No such file or directory

Work dir:
  /home/nelsonjs/tmp/repro-micromamba/test-repro/921e3a8aa32efe22bd73fbe1cbdfd09b/work/e2/667fd3bde76e5bdf8df569044e15a6

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
```

### Environment 

* Nextflow version: 23.04.2 
* Java version: java 11.0.16 2022-07-19 LTS
* Operating system: Linux 
* Bash version: 4.2.46
* Micromamba version: 1.4.7

### Additional context

I mentioned this in slack last week, but did not get any attention.  https://nextflow.slack.com/archives/C02T98A23U7/p1690325743801519

As near as I can tell from examining the source code, this configuration option cannot work, nor can the `useMamba` option, as nextflow always emits `conda info --json` for conda activation.

You may also want to add `--yes` to the `micromamba create` command, because if `export MAMBA_ALWAYS_YES=true` is not set, environment creation will hang indefinitely.

### PR DESCRIPTION
**Pull request by bentsherman**
Close #4161 

Tested with micromamba image:
```console
$ docker run --rm -it mambaorg/micromamba:1.4.9 -- bash
(base) $ command -v conda && (conda info --json | awk '/conda_prefix/ { gsub(/"|,/, "", $2); print $2 }') || (micromamba info | grep "env location :" | cut -d : -f 2)
 /opt/conda
```

### CODE REVIEW

**Comment by jemma-nelson** [comment]
Note that this PR only addresses the `micromamba info` invocation in point 2 of this comment: 
https://github.com/nextflow-io/nextflow/issues/4161#issuecomment-1709028041

Another change would required to address point 1, the `micromamba create --yes` flag.

(I'm neutral on how y'all handle this, just want to make sure its clear).

**Comment by pditommaso** [comment]
We are going to deprecate the support for Micromamba https://github.com/nextflow-io/nextflow/issues/4464. Therefore merging w/o closing

**Comment by ewels** [comment]
Reopening, as not sure we should remove Mamba just yet, at least until https://github.com/conda/conda/issues/13037 is solved.

**Comment by pditommaso** [comment]
@jorgee avoid the fallback approach can be a mess to debug, there should be a separate command when the micromamba flag is set

**Comment by jorgee** [comment]
> @jorgee avoid the fallback approach can be a mess to debug, there should be a separate command when the micromamba flag is set

I have pushed new changes with this modification. I think it is ready

File: modules/nextflow/src/test/groovy/nextflow/executor/BashWrapperBuilderTest.groovy [hunk 1]
```diff
@@ -775,13 +777,20 @@ class BashWrapperBuilderTest extends Specification {
...
+        where:
+        MICROMAMBA | RESULT
+        false      | '''\
                 # conda environment
-                source $(conda info --json | awk '/conda_prefix/ { gsub(/"|,/, "", $2); print $2 }')/bin/activate /some/conda/env/foo
+                source $(conda info --json | awk \'/conda_prefix/ { gsub(/"|,/, "", $2); print $2 }\')/bin/activate /some/conda/env/foo
+                '''.stripIndent()
+        true       | '''\
+                # conda environment
+                eval "$(micromamba shell hook --shell bash)" && micromamba activate /some/conda/env/foo
```
Comment (pditommaso) [inline modules/nextflow/src/test/groovy/nextflow/executor/BashWrapperBuilderTest.groovy:792]: better to add a new test for micromamba and keep the one for conda as it is

File: modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy [hunk 2]
```diff
@@ -285,7 +285,8 @@ class CondaCache {
         def cmd
         if( isYamlFilePath(condaEnv) ) {
             final target = isYamlUriPath(condaEnv) ? condaEnv : Escape.path(makeAbsolute(condaEnv))
-            cmd = "${binaryName} env create --prefix ${Escape.path(prefixPath)} --file ${target}"
+            final yesOpt = binaryName == 'micromamba' ? '--yes ' : ''
+            cmd = "${binaryName} env create ${yesOpt}--prefix ${Escape.path(prefixPath)} --file ${target}"
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy:289]: Let's add a test capturing the command for micromamba

**Review by pditommaso — CHANGES_REQUESTED** [review]

File: modules/nextflow/src/test/groovy/nextflow/executor/BashWrapperBuilderTest.groovy (hunk 1 above)
Comment (jorgee) [reply modules/nextflow/src/test/groovy/nextflow/executor/BashWrapperBuilderTest.groovy]: Modified in last commit

File: modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy (hunk 2 above)
Comment (jorgee) [reply modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy:289]: Added in last commit

**Pull request merged**